src/commands/publish_images.yml

```diff
@@ -7,17 +7,22 @@ parameters:
     type: string
   organization:
     description: Registry organization
-    type: string
+    type: env_var_name
   login_variable:
     description: Environment variable holding the registry login
     type: env_var_name
   password_variable:
     description: Environment variable holding the registry password
     type: env_var_name
+  set-variables:
+    description: "Steps that will be executed before running docker commands"
+    type: steps
+    default: []
```

Comment:
Do we really need this? Similar to my previous comment, I'd rather see additional parameters if they are required with default values that map what's in `Makefile.common`.

Reply:
We need to build and push more than one image which have different Dockerfile paths and image names. We will be using this step to set below values. 

```
set-variables:
          - run: echo "export DOCKERFILE_PATH=./cmd/prombench/Dockerfile" >> $BASH_ENV
          - run: echo "export DOCKER_IMAGE_NAME=prombench" >> $BASH_ENV
```

If we were to set these in the `Makefile.common`, it will still need the image name to be passed as a variable to determine the dockerfile path. 

I have set the default value to `[]`. So it functions in the same way as it currently does by reading values from `Makefile.common`.
Otherwise, it runs the specified steps.

Reply:
My suggestion would be to use conditional steps though I haven't tested them thoroughly.

```
parameters:
  registry:
    description: Registry address
    type: string
  organization:
    description: Registry organization
    type: string
  dockerfile_path:
    description: Path to Dockerfile
    type: string
    default: ""
  container_image_name:
    description: Name of the container image
    type: string
    default: ""
  login_variable:
    description: Environment variable holding the registry login
    type: env_var_name
  password_variable:
    description: Environment variable holding the registry password
    type: env_var_name

steps:
- when:
    conditions: << parameters.dockerfile_path >>
    steps:
    - run: echo "export DOCKERFILE_PATH=<< parameters.dockerfile_path >>" >> $BASH_ENV
- when:
    conditions: << parameters.container_image_name >>
    steps:
    - run: echo "export DOCKER_IMAGE_NAME=<< parameters.container_image_name >>" >> $BASH_ENV
...
```

From Circle CI orbs [documentation](https://circleci.com/docs/2.0/reusing-config/#defining-conditional-steps):

> The empty string will resolve as falsey in when conditions.

Meaning that existing jobs (eg prometheus, alertmanager, ...) wouldn't execute any of the pre-steps and the job would still use the values from `Makefile`.

Reply:
Done